py/emitinlinerv32.c

```diff
@@ -785,10 +817,11 @@ static void emit_inline_rv32_opcode(emit_inline_asm_t *emit, qstr opcode, mp_uin
         }
     }
 
-    if (!opcode_data) {
-        emit_inline_rv32_error_exc(emit,
-            mp_obj_new_exception_msg_varg(&mp_type_SyntaxError,
-                MP_ERROR_TEXT("unknown RV32 instruction '%q'"), opcode));
+    if (!opcode_data ||
+        (((opcode_data->required_extensions & ALLOWED_EXTENSIONS) != opcode_data->required_extensions) &&
+         ((opcode_data->required_extensions & emit_inline_rv32_allowed_extensions) != opcode_data->required_extensions))) {
```

Comment:
As suggested above, this could be simplified to `opcode_data->required_extensions & asm_rv32_allowed_extensions() != opcode_data->required_extensions`.